### ISSUE TITLE
Needs updating!

### ISSUE THREAD

**Issue by BogdanCerovac** [BogdanCerovac/aXeSiA#40]
I've obviously neglected this repo a bit but it is serving me well as is.
Should be updated and it's dependencies upgraded as the first prio.
Secondarily I was starting to use Playwright on some other test projects and got very good results with it - especially because it makes use of Firefox and Webkit possible. TBD...
Storage can also be a bottleneck with abundance of audits, so maybe switching to more scalable solutions... TBD

**Comment by Jym77** [comment]
Hi @BogdanCerovac,
just noticing this repo now.

I'm the maintainer of Alfa at Siteimprove. I obviously love that you are using it 😄 

I wanted to mention that we're releasing an [`alfa-test-utils` package](https://github.com/Siteimprove/alfa-integrations/tree/main/packages/alfa-test-utils) which is providing some wrappers for calling Alfa and logging results. It may make your life easier (or maybe not, depends on how many custom things you do…) See also [its documentation](https://alfa.siteimprove.com/code-checker).

Among others things, we took the opportunity to publish the packages in the classic npmjs registry (not just the Github one) which is much easier to connect to.

Also, I've noticed you're doing some time measurement in your code. It is possible to pass a performance hook to the `Audit#run` method (automatically done by the test utils package) to get some detailed timing on each rule.

(I thought it was OK to use this issue for that info since it relates to newest updates to Alfa)

---
You may also be interested in [Testaro](https://www.npmjs.com/package/testaro/v/48.0.2) which seem to have similar goals as aXeSiA in using several tools for testing the same page.

**Comment by BogdanCerovac** [comment]
Hi Jean-Yves (@Jym77), thanks for Alfa and thanks for this message 😄.

Will definitely look into wrappers and thank you for making it a bit more accessible (pun intended) with the npmjs registry.

I am not actively using this project anymore, had mostly used it to compare detection levels on different pages with both engines and to also check what can be done with merging the results and detecting the "most probably worse pages"...

Besides time measurement that you mentioned, are there any other possibilities to measure other relevant timings (like for example measurement of accessibility tree generation)?
I've tried to get that info in this Proof of concept: https://github.com/BogdanCerovac/POC-a11y-Accessibility.loadComplete but yes, probably out of the scope for Alfa....

I've seen the impressive Testaro project after reading the academic article behind it, but unfortunately didn't get the chance to really test it out yet... 1006 tests surely looks impressive 🤓.

**Comment by Jym77** [comment]
> Besides time measurement that you mentioned, are there any other possibilities to measure other relevant timings (like for example measurement of accessibility tree generation)? I've tried to get that info in this Proof of concept: https://github.com/BogdanCerovac/POC-a11y-Accessibility.loadComplete but yes, probably out of the scope for Alfa....

Yes. Alfa is in a weird position on that since we do build our own DOM, CSSOM and ARIA tree. So measuring the timings is easy ([here's how we do it in the test-utils package](https://github.com/Siteimprove/alfa-integrations/blob/main/packages/alfa-test-utils/src/audit/audit.ts#L245-L256)). But also disconnected from what happens in actual browsers, so probably less valuable.
